types/validator_set.go

```diff
@@ -270,34 +278,21 @@ func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID, height i
 	}
 
 	talliedVotingPower := int64(0)
-	round := commit.Round()
+	commit.AddAddresses(vals.GetAddresses())
 
 	for idx, precommit := range commit.Precommits {
 		if precommit == nil {
 			continue // OK, some precommits can be missing.
 		}
-		if precommit.Height != height {
```

Comment:
shouldn't we still check this once for `commit`?

Reply:
We do check that the commit has height/blockID which match the function parameters, but that code is outside the context here. https://github.com/tendermint/tendermint/blob/e5965ababe944d7470b4c21c52d09af93a8640b6/types/validator_set.go#L272-L283